src/runtime/tasker_noos.go

```diff
@@ -75,8 +75,10 @@ import (
 // Tasker code does not use FPU so the architecture specific context switch
 // code can avoid saving/restoring FPU context if not need.
 
+var dummyNanoseconds int64
+
 //go:nosplit
-func dummyNanotime() int64 { return 1 }
+func dummyNanotime() int64 { dummyNanoseconds += 100; return dummyNanoseconds }
```

Comment:
Use Xadd64.

To think about:

1. How it works with the rtos.SetSystemTimer and implementations. With the current code it may probably cause move backward in time. SetSystemTimer needs review. Can it be used multiple times? Etc...

2. How it works with 32-bit architectures where 64-bit "atomics" use spinlocks. Is it always safe?

Reply:
I'm not sure about this anymore. It will only help if curcpuSleep is a noop, which is something I want to change. I probably can't put the N64's CPU to sleep without messing other components up, but I would like to stall it at least and use the Systemtimer. Maybe it's not worth the extra complexity it will add to SetSystemTimer

Reply:
I thought about this a bit and even though I liked this fake timer at first I think it's not a good idea.

1. It may make difficult to catch the moment when runtime start require proper time progress during initialization.
2.  It's adds a lot of uncertainty and unpredictability.

In my opinion if some code require some kind of delay it should wait until the proper timer is configured. I'd like to stick with this stopped time at boot as long as it doesn't seriously harm anything.

Did you really require this fake timer at the very beginning of boot?

Reply:
There is runtime.usleep(), which is used very rarely. But it is used in freezetheworld(), which caused unprinted panics if raised before systimer setup. I often had this problem in init() of imported packages that aren't fully supported by embeddedgo. It made me backtracing manually which was a bit annoying, but it could also be confusing to newcomers.

I think the better solution might be change runtime.usleep() to not rely on the systimer. See [runtime/os_noos.go#L85](https://github.com/embeddedgo/go/blob/master-embedded/src/runtime/os_noos.go#L85)